OpcUaClient/OpcUaInterface.hpp

```diff
@@ -255,9 +265,174 @@ namespace Umati {
+    				UA_MonitoredItemCreateRequest_init(&item);
+					item.itemToMonitor.attributeId = UA_ATTRIBUTEID_EVENTNOTIFIER;
+					item.monitoringMode = UA_MONITORINGMODE_REPORTING;
+    				item.itemToMonitor.nodeId = UA_NODEID_NUMERIC(0, 2253); // Root->Objects->Server
+    			
+					UA_EventFilter filter;
+    				UA_EventFilter_init(&filter);
+
+					//Setup selection Clauses for Filter
+					int nSelectClauses = 3;
+					UA_SimpleAttributeOperand *selectClauses = (UA_SimpleAttributeOperand*)
+        			UA_Array_new(nSelectClauses, &UA_TYPES[UA_TYPES_SIMPLEATTRIBUTEOPERAND]);
+
+					for(size_t i =0; i<nSelectClauses; ++i) {
+        				UA_SimpleAttributeOperand_init(&selectClauses[i]);
+    				}
+					selectClauses[0].typeDefinitionId = UA_NODEID_NUMERIC(0, UA_NS0ID_BASEEVENTTYPE);
+					selectClauses[0].attributeId = UA_ATTRIBUTEID_VALUE;
+    				selectClauses[0].browsePathSize = 1;
+    				selectClauses[0].browsePath = (UA_QualifiedName*)
+       				UA_Array_new(selectClauses[0].browsePathSize, &UA_TYPES[UA_TYPES_QUALIFIEDNAME]);
+    				if(!selectClauses[0].browsePath) {
+        				UA_SimpleAttributeOperand_delete(selectClauses);
+    				}
+    				selectClauses[0].browsePath[0] = UA_QUALIFIEDNAME_ALLOC(0, "EventType");
+					selectClauses[1].typeDefinitionId = UA_NODEID_NUMERIC(0, UA_NS0ID_BASEEVENTTYPE);
+					selectClauses[1].attributeId = UA_ATTRIBUTEID_VALUE;
+    				selectClauses[1].browsePathSize = 1;
+    				selectClauses[1].browsePath = (UA_QualifiedName*)
+       				UA_Array_new(selectClauses[1].browsePathSize, &UA_TYPES[UA_TYPES_QUALIFIEDNAME]);
+
+    				if(!selectClauses[1].browsePath) {
+        				UA_SimpleAttributeOperand_delete(selectClauses);
+    				}
+    				selectClauses[1].browsePath[0] = UA_QUALIFIEDNAME_ALLOC(0, "SourceNode");
+
+					selectClauses[2].typeDefinitionId = UA_NODEID_NUMERIC(0, UA_NS0ID_BASEEVENTTYPE);
+					selectClauses[2].attributeId = UA_ATTRIBUTEID_VALUE;
+    				selectClauses[2].browsePathSize = 1;
+    				selectClauses[2].browsePath = (UA_QualifiedName*)
+       				UA_Array_new(selectClauses[2].browsePathSize, &UA_TYPES[UA_TYPES_QUALIFIEDNAME]);
+    				if(!selectClauses[2].browsePath) {
+        				UA_SimpleAttributeOperand_delete(selectClauses);
+    				}
+
+    				selectClauses[2].browsePath[0] = UA_QUALIFIEDNAME_ALLOC(0, "Changes");    
+
+					filter.selectClauses = selectClauses;
+    				filter.selectClausesSize = 3;
+					// Content Filter
+					UA_ContentFilter contentFilter;
+					UA_ContentFilter_init(&contentFilter);
+
+					UA_ContentFilterElement contentFilterElement;
+					UA_ContentFilterElement_init(&contentFilterElement);
+
+					UA_LiteralOperand literalOperand;
+					UA_LiteralOperand_init(&literalOperand);
+					UA_Variant v;
+					UA_NodeId generalModelChangeEventNodeId = UA_NODEID_NUMERIC(0, UA_NS0ID_GENERALMODELCHANGEEVENTTYPE);
+					UA_Variant_setScalar(&v, &generalModelChangeEventNodeId, &UA_TYPES[UA_TYPES_NODEID]);
```

Comment:
Rejected

Reply:
I'm unsure about the implementation of `UA_Variant_setScalar`, but if this function does clear the existing value first, UA_Variant must be initialied, to prevent random memory access.

Reply:
Thanks for clearification. I will lookup that.